recipes/lora_finetune_distributed.py

```diff
@@ -477,6 +477,19 @@ def save_checkpoint(
         # to be sent to the checkpointer and ultimately written to file
         if self._is_rank_zero:
 
+            # if training is in-progress, checkpoint the optimizer state and recipe state
```

Comment:
Dumb q: Why move this up?

Reply:
Mainly to align with the ordering in the single-device recipe, but realistically it doesn't matter too much. Maybe I'll just leave as-is to not mix in extra complexity with the current set of changes. Also I realized somehow my changes to save PEFT config did not get pushed to this recipe, will update that now.